### PR TITLE
ENH: Add binary tissue classifier option for tracking workflow

### LINKED ISSUE #1811: Add binary tissue classifier option for the tracking workflow
Workflow: **dipy_track_local**

A threshold tissue classifier with parameter --stopping_thr is currently used. It would be nice to offer either a threshold tissue classifier or a binary tissue classifier here. With the mask being: stopping_file > stopping_thr.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add binary tissue classifier option for tracking workflow.

Fixes #1811.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@gabknight reviews are welcome.

File: dipy/workflows/tests/test_tracking.py [hunk 1]
```diff
@@ -143,6 +143,16 @@ def test_local_fiber_tracking_workflow():
             lf_track_pam.last_generated_outputs['out_tractogram']
         assert_false(is_tractogram_empty(tractogram_path))
 
+        # Test tracking with binary tissue classifier
+        lf_track_pam = LocalFiberTrackingPAMFlow()
+        lf_track_pam._force_overwrite = True
+        lf_track_pam.run(pam_path, gfa_path, seeds_path,
```
Comment (gabknight) [inline dipy/workflows/tests/test_tracking.py]: Although this is fine for testing, using a binary tissue classifier on the GFA will result in using a mask with 1s everywhere as the GFA > 0 in all voxels. I suggest we use the threholded gfa mask defined for the seeding instead.

Here replacing `gfa_path`, with `mask_path` and setting at line 131:
```mask_path = mask_flow.last_generated_outputs['out_mask']```

File: dipy/workflows/tracking.py [hunk 2]
```diff
@@ -134,6 +139,8 @@ def run(self, pam_files, stopping_files, seeding_files,
             Path to images (e.g. FA) used for stopping criteria for tracking.
         seeding_files : string
             A binary image showing where we need to seed for tracking.
+        use_binary_mask : bool, optional
+            If True, uses a binary tissue classifier.
```
Comment (gabknight) [inline dipy/workflows/tracking.py]: Maybe add here:
"If the provided `stopping_files` are not binary, the `stopping_thr` will be used to binarize the images.

**Comment by gabknight** [comment]
See my 2 minor comments above. Otherwise, this LGTM.

Thanks for doing this PR @jhlegarreta

File: dipy/workflows/tracking.py (hunk 2 above)
Comment (jhlegarreta) [reply dipy/workflows/tracking.py]: Thanks for the review @gabknight. Done in 32ba095.

File: dipy/workflows/tests/test_tracking.py (hunk 1 above)
Comment (jhlegarreta) [reply dipy/workflows/tests/test_tracking.py]: Thanks for the review @gabknight. Done in b3a4b27.

**Comment by gabknight** [comment]
Thanks @jhlegarreta 

Let see if Travis has something to say. Otherwise, this seems ready to be merged.

**Comment by jhlegarreta** [comment]
192fb69 rebased on master.

**Pull request merged**